primer/brand
# Show VideoPlayer tooltips when associated control receives focus

## Summary

Previously, the `VideoPlayer` component's tooltips would only show when the associated control was hovered. This PR makes them show on focus as well.

Unfortunately we can't just use the `:focus` pseudo-class because we need to be able to dismiss the tooltip with the <kbd>Esc</kbd> key.

Longer term, we'd benefit from using the PB `Tooltip` component, rather than this custom implementation.

This change doesn't show the tooltip when focusing the `Range` sliders (eg seek, volume) as those tooltips are currently bound to mouse position, which isn't relevant when using keyboard controls.

## List of notable changes:

- Show/hide `VideoTooltip` on focus/blur
- Allow dismissing with <kbd>Esc</kbd> key
- Remove duplicate labels from some of the controls
- Reduce scope of mousemove listener. Previously it listened for mousemoves on the whole window, which was unnecessary.

## What should reviewers focus on?

- Check that tooltips appear when the relevant control is focused, and that they can be dismissed with <kbd>Esc</kbd>

## Steps to test:

1. Open one of the `VideoPlayer` stories
1. Tab through the video's controls
1. Observe that the Play, CC, Mute, and Fullscreen buttons display tooltips when they receive focus
1. Observe that the tooltips can be dismissed with the <kbd>Esc</kbd> key

## Supporting resources (related issues, external links, etc):

- Closes https://github.com/github/primer/issues/3804

## Contributor checklist:


## Reviewer checklist:
